src/DevBetterWeb.Web/Controllers/PaymentsController.cs

```diff
@@ -0,0 +1,277 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using DevBetterWeb.Infrastructure.Services;
+using DevBetterWeb.Web.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using Stripe;
+using Stripe.Checkout;
+
+
+namespace DevBetterWeb.Web.Controllers
+{
+  public class PaymentsController : Controller
+  {
+    public readonly IOptions<StripeOptions> options;
+    private readonly IStripeClient client;
+
+    public PaymentsController(IOptions<StripeOptions> options)
+    {
+      this.options = options;
+      client = new StripeClient(this.options.Value.stripeSecretKey);
+    }
+
+    // Set your secret key. Remember to switch to your live secret key in production!
+    // See your keys here: https://dashboard.stripe.com/account/apikeys
+
+    [HttpGet("setup")]
+    public SetupResponse Setup()
+    {
+      return new SetupResponse
+      {
+        //ProPrice = this.options.Value.ProPrice,
+        //BasicPrice = this.options.Value.BasicPrice,
+        PublishableKey = this.options.Value.stripePublishableKey,
+      };
+    }
+
+    [HttpPost("create-checkout-session")]
+    public async Task<IActionResult> CreateCheckoutSession([FromBody] CreateCheckoutSessionRequest req)
+    {
+      var options = new SessionCreateOptions
+      {
+        // See https://stripe.com/docs/api/checkout/sessions/create
+        // for additional parameters to pass.
+        // {CHECKOUT_SESSION_ID} is a string literal; do not change it!
+        // the actual Session ID is returned in the query parameter when your customer
+        // is redirected to the success page.
+        SuccessUrl = "https://example.com/success.html?session_id={CHECKOUT_SESSION_ID}",
+        CancelUrl = "/checkout",
+        PaymentMethodTypes = new List<string>
+        {
+            "card",
+        },
+        Mode = "subscription",
+        LineItems = new List<SessionLineItemOptions>
+        {
+            new SessionLineItemOptions
+            {
+                Price = req.PriceId,
+                // For metered billing, do not pass quantity
+                Quantity = 1,
+            },
+        },
+      };
+      var service = new SessionService(this.client);
+      try
+      {
+        var session = await service.CreateAsync(options);
+        return Ok(new CreateCheckoutSessionResponse
+        {
+          SessionId = session.Id,
+        });
+      }
+      catch (StripeException e)
+      {
+        Console.WriteLine(e.StripeError.Message);
```

Comment:
This should be using a logger, not Console.WriteLine. You can inject `ILogger<PaymentsController>` in this class's constructor and assign it to a field (`_logger`) and use `_logger` here.